Source/WebCore/platform/calc/CalcOperator.h

```diff
@@ -63,4 +65,239 @@ enum class CalcOperator : uint8_t {
+        double upperB = lowerB + std::abs(b);
+        return std::make_pair(lowerB, upperB);
+    };
+
+    switch (calcOperator) {
+    case CalcOperator::Add: {
+        double sum = 0;
+        for (auto& child : children)
+            sum += evaluate(child);
+        return sum;
+    }
+    case CalcOperator::Subtract:
+        ASSERT(children.size() == 2);
+        return evaluate(children[0]) - evaluate(children[1]);
+    case CalcOperator::Multiply: {
+        double product = 1;
+        for (auto& child : children)
+            product *= evaluate(child);
+        return product;
+    }
+    case CalcOperator::Divide:
+        ASSERT(children.size() == 1 || children.size() == 2);
+        if (children.size() == 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return evaluate(children[0]) / evaluate(children[1]);
+    case CalcOperator::Min: {
+        if (children.isEmpty())
+            return std::numeric_limits<double>::quiet_NaN();
+        auto minimum = evaluate(children[0]);
+        for (auto& child : children) {
+            auto evaluatedChild = evaluate(child);
+            if (std::isnan(evaluatedChild))
+                return evaluatedChild;
+            minimum = std::min(minimum, evaluatedChild);
+        }
+        return minimum;
+    }
+    case CalcOperator::Max: {
+        if (children.isEmpty())
+            return std::numeric_limits<double>::quiet_NaN();
+        auto maximum = evaluate(children[0]);
+        for (auto& child : children) {
+            auto evaluatedChild = evaluate(child);
+            if (std::isnan(evaluatedChild))
+                return evaluatedChild;
+            maximum = std::max(maximum, evaluatedChild);
+        }
+        return maximum;
+    }
+    case CalcOperator::Clamp: {
+        if (children.size() != 3)
+            return std::numeric_limits<double>::quiet_NaN();
+        double min = evaluate(children[0]);
+        double value = evaluate(children[1]);
+        double max = evaluate(children[2]);
+        if (std::isnan(min) || std::isnan(value) || std::isnan(max))
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::max(min, std::min(value, max));
+    }
+    case CalcOperator::Pow:
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::pow(evaluate(children[0]), evaluate(children[1]));
+    case CalcOperator::Sqrt: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::sqrt(evaluate(children[0]));
+    }
+    case CalcOperator::Hypot: {
+        if (children.isEmpty())
+            return std::numeric_limits<double>::quiet_NaN();
+        if (children.size() == 1)
+            return std::abs(evaluate(children[0]));
+        double sum = 0;
+        for (auto& child : children) {
+            auto evaluatedChild = evaluate(child);
+            if (std::isnan(evaluatedChild))
```

Comment:
Not new: We can probably remove this without any effect on the results because of how NaN works. Would slightly speed up non-NaN cases and slow down NaN cases. Not important to change.

Reply:
Removed.